frontend/src/metabase/query_builder/components/view/ViewHeader/ViewHeader.jsx

```diff
@@ -352,14 +348,15 @@ function AhHocQuestionLeftSide(props) {
   );
 }
 
-DatasetCollectionBadge.propTypes = {
-  dataset: PropTypes.object.isRequired,
+HeaderCollectionBadge.propTypes = {
+  entity: PropTypes.object.isRequired,
```

Comment:
This is actually an instance of `Question`

```suggestion
  question: PropTypes.object.isRequired,
```